### PR TITLE
Hr contact crud

### LINKED ISSUE #80: CRUD + Schema validation for hr_contact
CRUD + Schema validation for hr_contact. See Wiki for details.

Endpoints under hr_contact controller:
- GET /api/hr_contact/{hr_contact_id}/
- UPDATE /api/hr_contact/{hr_contact_id}/
- DELETE /api/hr_contact/{hr_contact_id}/

**Note:** Below is kind of weird. Not certain if best way.

Endpoints under Employer controller (Because you only ever make a hr_contact that is for an employer):
- POST /api/employer/{employer_id}/hr_contact/

### PR DESCRIPTION
**Pull request by ganeshasapu**
# Description
* Note I also updated the http status codes for when we don't find an object in a PUT request. Link to the issue: https://github.com/uoftblueprint/tces/pull/82#discussion_r1095313308

Fixes #80
* Creates CRUD for hr_contact object
* Adds validation for hr_contact object
___
### Hr_Contact - POST
curl --location --request POST 'http://localhost:3001/api/employer/63dd48268061918c52a2add9/hr_contact' \
--header 'Content-Type: application/json' \
--data-raw '{
    "first_name": "Test2",
    "job_title": "software",
    "last_name": "smith",
    "phone_num": "12321451",
    "email": "Test@skadk.com"
}'

* **Note that 63dd48268061918c52a2add9 is ID of an employer**
*  **the hr_contacts array of that employer has been updated with the created hr_contact's ID (This can be verified using Employer - GET)**

### Response
Successfully created a new hr contact with id 63dd497f1451c4ee2471153a
___
### Hr_Contact - GET
curl --location --request GET 'http://localhost:3001/api/hr_contact/63dd497f1451c4ee2471153a'

### Response
{
    "_id": "63dd497f1451c4ee2471153a",
    "first_name": "Test2",
    "job_title": "software",
    "last_name": "smith",
    "phone_num": "12321451",
    "email": "Test@skadk.com",
    "alt_phone_num": null
}
___
### Hr_Contact - DELETE
curl --location --request DELETE 'http://localhost:3001/api/hr_contact/63dd497f1451c4ee2471153a'

### Response
Successfully removed hr contact with id 63dd497f1451c4ee2471153a
___
### Hr_Contact - PUT
curl --location --request PUT 'http://localhost:3001/api/hr_contact/63dd497f1451c4ee2471153a' \
--header 'Content-Type: application/json' \
--data-raw '{
    "first_name": "Tes232132t2",
    "job_title": "software",
    "last_name": "smith",
    "phone_num": "12321451",
    "email": "Test@skadk.com"
}'

### Response
Successfully updated hr contact with id 63dd497f1451c4ee2471153a
___
## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

**Review by helen-li — APPROVED** [review]
Slay! lgtm could you quickly rename `employer.ts` to `employer.model.ts` just to be consistent

**Review by jordanjanakievski — APPROVED** [review]
Everything worked. Just address Helen's file name change and you are good to go

File: server/src/controllers/client.controller.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ const updateClient = async (req: Request, res: Response, next: NextFunction) =>
     } catch (error) {
         if (error instanceof Error) {
             console.log(error);
-            res.status(304).send(error.message);
+            res.status(404).send(error.message);
```
Comment (kenneth-miura) [inline server/src/controllers/client.controller.ts:66]: this is actually fine, plz revert back to 404

File: server/src/controllers/contact.controller.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ const updateContact = async (
   } catch (e) {
     if (e instanceof Error) {
       error(e.message);
-      res.status(304).send(e.message);
+      res.status(404).send(e.message);
```
Comment (kenneth-miura) [inline server/src/controllers/contact.controller.ts:63]: this is actually fine, plz revert back to 404

File: server/src/controllers/hr_contact.controller.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const id = req?.params?.id;
+    const hrContact = await Hr_Contact.getHrContact(id);
+    res.status(200).json(hrContact);
+  } catch (error) {
+    if (error instanceof Error) {
+      console.log(error);
```
Comment (kenneth-miura) [inline server/src/controllers/hr_contact.controller.ts:15]: you should use logger

Comment (kenneth-miura) [reply server/src/controllers/hr_contact.controller.ts:15]: logger.error

Comment (kenneth-miura) [reply server/src/controllers/hr_contact.controller.ts:15]: for all these spots

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
looks good! See attached comments

**Review by kenneth-miura — APPROVED** [review]
lgtm!

**Pull request merged**